### PR TITLE
Feature/issue 1481 client upgrade

### PR DESCRIPTION
**Pull request by ian-lin-PG**

### CODE REVIEW

File: src/components/search/search_input_component.tsx [hunk 1]
```diff
@@ -309,26 +309,28 @@ const MyLocationButton = (props: {
...
+        case 'always':
+            setSearchLocationInput(i18n._(MY_LOCATION_MESSAGE_DESCRIPTOR));
+            break;
+        case 'fine':
+            setSearchLocationInput(i18n._(MY_LOCATION_MESSAGE_DESCRIPTOR));
+            break;
+        case 'coarse':
+            setSearchLocationInput(i18n._(MY_LOCATION_MESSAGE_DESCRIPTOR));
+            break;
+        case 'none':
```
Comment (rasmus-storjohann-PG) [inline src/components/search/search_input_component.tsx]: Shouldn't this be instead
```ts
if (status === 'none') {
    openAppSettings();
} else {
    setSearchLocationInput(i18n._(MY_LOCATION_MESSAGE_DESCRIPTOR));
}
```

File: src/components/search/use_ask_location_permission.ts [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
+import { isAndroid } from '../../application/helpers/is_android';
+import { requestLocationPermission } from '../../application/helpers/request_location_permission';
 
 export const useAskLocationPermission = (setLocationToMyLocation: () => void): void => {
     useEffect((): void => {
         const askLocationPermission = async (): Promise<void> => {
-            const getPermissionResponse = await Permissions.getAsync(Permissions.LOCATION);
-            const haveNeverAskedBefore = getPermissionResponse.status === Permissions.PermissionStatus.UNDETERMINED;
+            const getPermissionResponse = await Location.getPermissionsAsync();
+            const status = isAndroid() ? getPermissionResponse.android.scope : getPermissionResponse.ios.scope;
```
Comment (rasmus-storjohann-PG) [inline src/components/search/use_ask_location_permission.ts]: This is repeated from above, can be DRYed up

File: src/components/search/search_input_component.tsx [hunk 3]
```diff
@@ -311,19 +311,10 @@ const MyLocationButton = (props: {
...
-            setSearchLocationInput(i18n._(MY_LOCATION_MESSAGE_DESCRIPTOR));
-            break;
-        case 'fine':
-            setSearchLocationInput(i18n._(MY_LOCATION_MESSAGE_DESCRIPTOR));
-            break;
-        case 'coarse':
-            setSearchLocationInput(i18n._(MY_LOCATION_MESSAGE_DESCRIPTOR));
-            break;
-        case 'none':
+        case PermissionStatus.DENIED:
```
Comment (rasmus-storjohann-PG) [inline src/components/search/search_input_component.tsx:317]: If/else still the preferred form here

**Review by brandon-PG — APPROVED** [review]
Looks good to me

**Review by rasmus-storjohann-PG — CHANGES_REQUESTED** [review]
Really nice work, just some minor changes to suggest

**Comment by rasmus-storjohann-PG** [comment]
@ian-lin-PG one more thing, can you update to use eslint and https://github.com/jonaskello/eslint-plugin-functional while you're at it?

File: src/components/search/search_input_component.tsx (hunk 1 above)
Comment (ian-lin-PG) [reply src/components/search/search_input_component.tsx]: I eventually installed the expo-location package separately which has the more recognizable Permission.Status that we are more used to.

File: src/components/search/use_ask_location_permission.ts (hunk 2 above)
Comment (ian-lin-PG) [reply src/components/search/use_ask_location_permission.ts]: Updated package as described as above.

**Pull request merged**